raiden/tests/integration/api/test_pythonapi_regression.py

```diff
@@ -33,26 +33,29 @@ def test_close_regression(raiden_network, deposit, token_addresses):
     amount = PaymentAmount(10)
     identifier = PaymentID(42)
     secret, secrethash = factories.make_secret_with_hash()
-    assert api1.transfer_and_wait(
-        registry_address=registry_address,
-        token_address=token_address,
-        amount=amount,
-        target=TargetAddress(api2.address),
-        identifier=identifier,
-        secret=secret,
-        transfer_timeout=10,
-    )
-    exception = ValueError("Waiting for transfer received success in the WAL timed out")
-    with gevent.Timeout(seconds=5, exception=exception):
+    timeout = block_offset_timeout(app1.raiden)
+    with timeout:
```

Comment:
Here it also makes sense to have the `watch_for_unlock_failures`, since we are doing a transfer that we expect to succeed.

I think the best change is to have the context manager `watch_for_unlock_failures` watching for both cases, IOW, the  `block_offset_timeout` should also be used in there.

Reply:
`watch_for_unlock_failures` is used together with different timeouts and waiting functions, so I just added it in this place right now instead of changing it.